### DISCUSSION TITLE
Lesson 13: Module not found: Error: Can't resolve 'ethers'

### DISCUSSION THREAD

**Discussion by ace-brown** [smartcontractkit/full-blockchain-solidity-course-py#1586] [Q&A] [answered]
Hi guys. In the lesson 13 around  14:27:00  i faced an error.  Please help me.

`Failed to compile.

Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\abi\multicall'
WARNING in ./node_modules/@metamask/detect-provider/dist/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@metamask\detect-provider\src\index.ts' file: Error: ENOENT: no such file or directory, open 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@metamask\detect-provider\src\index.ts'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/abi/multicall/constants.js 1:0-31
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\abi\multicall'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/abi/multicall2/constants.js 1:0-31
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\abi\multicall2'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/constants/abi/index.js 1:0-31
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\constants\abi'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/helpers/address.js 1:0-31
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\helpers'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/helpers/calls.js 1:0-31
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\helpers'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/helpers/contract.js 1:0-41
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\helpers'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/helpers/getSignerFromOptions.js 1:0-32
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\helpers'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/helpers/injectedProvider.js 2:0-35
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\helpers'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/hooks/useBlockMeta.js 2:0-35
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\hooks'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/hooks/useEtherBalance.js 4:0-34
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\hooks'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/hooks/useEthers.js 1:0-46
Module not found: Error: Can't resolve 'ethers/lib/utils' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\hooks'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/hooks/usePromiseTransaction.js 3:0-43
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\hooks'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/hooks/useRawCalls.js 3:0-31
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\hooks'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/hooks/useToken.js 3:0-34
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\hooks'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/hooks/useTokenAllowance.js 1:0-34
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\hooks'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/hooks/useTokenBalance.js 1:0-34
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\hooks'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/model/CurrencyValue.js 1:0-35
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\model'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/providers/LocalMulticallProvider.js 3:0-31
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\providers'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/providers/chainState/common/multicall.js 1:0-34
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\providers\chainState\common'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/providers/chainState/common/multicall2.js 1:0-34
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\providers\chainState\common'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/providers/network/network/provider.js 5:0-35
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\providers\network\network'

ERROR in ./node_modules/@usedapp/core/dist/esm/src/providers/network/readonlyNetworks/provider.js 3:0-35
Module not found: Error: Can't resolve 'ethers' in 'D:\Media\Work\Web development\My projects\Blockchain projects\stake-and-yield-brownie\front_end\node_modules\@usedapp\core\dist\esm\src\providers\network\readonlyNetworks'     

webpack compiled with 22 errors and 1 warning
Files successfully emitted, waiting for typecheck results...
Issues checking in progress...
No issues found.`

**Comment by cromewar** [comment]
Hello @KAD105 
The web pack error is actually common, the solution is to use another node version, the latest LTS version 16.x should work

**Reply by ace-brown** [reply]
I just updated the node version, but it's raising the same error :(

**Comment by ace-brown** [comment, marked as answer]
However i managed to handle the error. I found out that it must be something about ethers, so with a bit google research i found this line of code:
> npm install --save ethers

and now it's solved!

**Reply by cromewar** [reply]
I'm glad your problem is solved now.

**Reply by ace-brown** [reply]
Thank you for spending time reading error in my codes. I appreciate that.

**Reply by Niferu** [reply]
Hi @ace-brown 
Had same issue, thanks a lot for solving it for me!

**Reply by ace-brown** [reply]
My pleasure :)
